src/terrain/util/oauth.clj

```diff
@@ -34,20 +35,26 @@
         (handler request)
         (resp/forbidden "You are not in one of the admin groups.")))))
 
+(defn get-oauth-profile
+  [token]
+  (try+
+   (doto (oauth/get-profile token)
+     (validate-oauth-profile))
+   (catch [:status 401] _
+     (log/warn (:throwable &throw-context) "Invalid or expired OAuth token.")
+     (resp/unauthorized "Invalid or expired OAuth token."))
+   (catch [:type :validation] _
+     (resp/forbidden (.getMessage (:throwable &throw-context))))
+   (catch Object _
+     (log/error (:throwable &throw-context) "Unexpected exception.")
+     (resp/error-response http-resp/internal-server-error
+                          ce/ERR_UNCHECKED_EXCEPTION
+                          "Received an unexpected exception."))))
+
 (defn validate-oauth-token
   [handler token-fn]
   (fn [request]
-    (try+
-     (if-let [token (token-fn request)]
-       (let [profile (oauth/get-profile token)]
-         (validate-oauth-profile profile)
-         (handler (assoc request :oauth-profile profile)))
-       (resp/unauthorized "No OAuth token found."))
-     (catch [:status 401] _
-       (log/warn (:throwable &throw-context) "Invalid or expired OAuth token.")
-       (resp/unauthorized "Invalid or expired OAuth token."))
-     (catch [:type :validation] _
-       (resp/forbidden (.getMessage (:throwable &throw-context))))
-     (catch Object _
-       (log/error (:throwable &throw-context) "Unexpected exception.")
-       (resp/error-response ce/ERR_UNCHECKED_EXCEPTION "Received an unexpected exception.")))))
+    (if-let [token (token-fn request)]
+      (let [profile (get-oauth-profile token)]
+        (handler (assoc request :oauth-profile profile)))
+      (resp/unauthorized "No OAuth token found."))))
```

Comment:
Nice refactor 👍

Reply:
Thanks. When I was updating it, I found myself wondering why I didn't write it this way in the first place. 😆